### PR TITLE
feat: add `metadata.label` to synced secret

### LINKED ISSUE #217: Enable add `metadata.label` to synced secret
**Describe the solution you'd like**
We would like to have a parameter to add the label to the synced secrets. Our applications are  using spring-cloud-kubernetes to retrievew the data from secrets, it needs the label to access the specified secret.
Also custom labels is useful to daily operation.

**Anything else you would like to add:**
The secretObjects doesn't have any label related parameters
```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: my-provider
spec:
  provider: azure                             
  secretObjects:                              # [OPTIONAL] SecretObject defines the desired state of synced K8s secret objects
  - data:
    - key: username                           # data field to populate
      objectName: foo1                        # name of the mounted content to sync. this could be the object name or the object alias
    secretName: foosecret                     # name of the Kubernetes Secret object
    type: Opaque                              # type of the Kubernetes Secret object e.g. Opaque, kubernetes.io/tls
```

**Environment:**

- Secrets Store CSI Driver version:  csi-node-driver-registrar:v1.2.0 , secrets-store-csi:v0.0.11
- Kubernetes version: v1.16.9

### PR DESCRIPTION
**Pull request by helayoty**
**What this PR does / why we need it**:

There was an ask to support label in synced secrets

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #217 

**Special notes for your reviewer**:

### CODE REVIEW

File: controllers/secretproviderclasspodstatus_controller.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func (r *SecretProviderClassPodStatusReconciler) Reconcile(req ctrl.Request) (ct
 			}
 
 			createFn := func() (bool, error) {
-				if err := r.createK8sSecret(ctx, secretObj.SecretName, req.Namespace, datamap, secretType); err != nil {
+				if err := r.createK8sSecret(ctx, secretObj.Labels, secretObj.SecretName, req.Namespace, datamap, secretType); err != nil {
```
Comment (aramase) [inline controllers/secretproviderclasspodstatus_controller.go:179]: Can we move `secretObj.Labels` to be after datamap in the function signature?

File: controllers/secretproviderclasspodstatus_controller.go [hunk 2]
```diff
@@ -226,11 +226,12 @@ func (r *SecretProviderClassPodStatusReconciler) SetupWithManager(mgr ctrl.Manag
 
 // createK8sSecret creates K8s secret with data from mounted files
 // If a secret with the same name already exists in the namespace of the pod, the error is nil.
-func (r *SecretProviderClassPodStatusReconciler) createK8sSecret(ctx context.Context, name, namespace string, datamap map[string][]byte, secretType corev1.SecretType) error {
+func (r *SecretProviderClassPodStatusReconciler) createK8sSecret(ctx context.Context, labelsmap map[string]string, name, namespace string, datamap map[string][]byte, secretType corev1.SecretType) error {
```
Comment (aramase) [inline controllers/secretproviderclasspodstatus_controller.go:229]:
```suggestion
func (r *SecretProviderClassPodStatusReconciler) createK8sSecret(ctx context.Context, name, namespace string, datamap map[string][]byte, labelsmap map[string]string, secretType corev1.SecretType) error {
```

File: controllers/secretproviderclasspodstatus_controller_test.go [hunk 3]
```diff
@@ -46,11 +46,12 @@ func setupScheme() (*runtime.Scheme, error) {
 	return scheme, nil
 }
 
-func newSecret(name, namespace string) *v1.Secret {
+func newSecret(labels map[string]string, name, namespace string) *v1.Secret {
```
Comment (aramase) [inline controllers/secretproviderclasspodstatus_controller_test.go:49]:
```suggestion
func newSecret(name, namespace string, labels map[string]string) *v1.Secret {
```

File: controllers/secretproviderclasspodstatus_controller_test.go [hunk 4]
```diff
@@ -136,20 +141,28 @@ func TestCreateK8sSecret(t *testing.T) {
...
 
-	err = reconciler.createK8sSecret(context.TODO(), "my-secret2", "default", nil, v1.SecretTypeOpaque)
+	err = reconciler.createK8sSecret(context.TODO(), labels, "my-secret2", "default", nil, v1.SecretTypeOpaque)
 	g.Expect(err).NotTo(HaveOccurred())
 	secret := &v1.Secret{}
+
 	err = client.Get(context.TODO(), types.NamespacedName{Name: "my-secret2", Namespace: "default"}, secret)
 	g.Expect(err).NotTo(HaveOccurred())
+
+	g.Expect(err).NotTo(HaveOccurred())
```
Comment (aramase) [inline controllers/secretproviderclasspodstatus_controller_test.go:163]: This can be removed as it's already checked in line 161

File: deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml [hunk 5]
```diff
@@ -61,6 +61,11 @@ spec:
                           type: string
                       type: object
                     type: array
+                  labels:
```
Comment (aramase) [inline deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml:64]: Try a rebase with master and then generating the manifests using `make manifests`. With #258 we have added `manifest_staging` dir to host the chart/deploy changes. These changes will be tagged with the release. Only manifests in `manifest_staging` need to be changed as part of the PR

File: test/bats/azure.bats [hunk 6]
```diff
@@ -121,6 +122,9 @@ setup() {
   result=$(kubectl get secret foosecret -o jsonpath="{.data.username}" | base64 -d)
   [[ "${result//$'\r'}" -eq "${SECRET_VALUE}" ]]
 
+  result=$(kubectl get secret foosecret -o jsonpath="{.metadata.labels[0]}")
```
Comment (aramase) [inline test/bats/azure.bats:125]: This command will error as labels is `map[string]interface{}`. It should be -

```suggestion
  result=$(kubectl get secret foosecret -o jsonpath="{.metadata.labels.environment}")
```

This will return the value of the label if exists

File: test/bats/azure.bats [hunk 7]
```diff
@@ -121,6 +122,9 @@ setup() {
...
+  [[ "${result//$'\r'}" -eq "${LABEL_VALUE}" ]]
```
Comment (aramase) [inline test/bats/azure.bats:126]: We need to add this test to `vault.bats` too

**Comment by helayoty** [comment]
remove /do-not-merge/work-in-progress

File: charts/secrets-store-csi-driver/templates/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml [hunk 8]
```diff
@@ -61,6 +61,11 @@ spec:
                           type: string
                       type: object
                     type: array
+                  labels:
```
Comment (aramase) [inline charts/secrets-store-csi-driver/templates/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml:64]: Can you revert the changes from `charts` and `deploy` dir. The CRD spec needs to be updated in `manifest_staging/deploy` and `manifest_staging/charts/secrets-store-csi-driver` only. The e2e tests are run against the helm charts in `manifests_staging` - https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/master/manifest_staging/charts/secrets-store-csi-driver

File: controllers/secretproviderclasspodstatus_controller_test.go (hunk 3 above)
Comment (aramase) [reply controllers/secretproviderclasspodstatus_controller_test.go:49]: This still needs to be updated?

File: test/bats/vault.bats [hunk 9]
```diff
@@ -184,6 +185,9 @@ EOF
   result=$(kubectl exec -it $POD -- printenv | grep SECRET_USERNAME | awk -F"=" '{ print $2 }' | tr -d '\r\n')
   [[ "$result" == "hello1" ]]
 
+  result=$(kubectl get secret foosecret -o jsonpath="{.metadata.labels.environment}")
+  [[ "${result//$'\r'}" -eq "${LABEL_VALUE}" ]]
```
Comment (aramase) [inline test/bats/vault.bats:189]: Use `==` instead of `-eq`

File: test/bats/azure.bats (hunk 7 above)
Comment (aramase) [reply test/bats/azure.bats:131]: Use `==` instead of `-eq`

File: test/bats/azure.bats [hunk 10]
```diff
@@ -123,6 +124,12 @@ setup() {
   result=$(kubectl get secret foosecret -o jsonpath="{.data.username}" | base64 -d)
   [[ "${result//$'\r'}" == "${SECRET_VALUE}" ]]
 
+  result=$(kubectl get secret foosecret -o jsonpath="{.metadata.labels.environment}")
```
Comment (aramase) [inline test/bats/azure.bats:127]: This is a duplicate of the same command. Could you remove it?

**Comment by aramase** [comment]
The windows test job template needs to be updated. Will re-run the windows test once that update is complete.
